lib/postzord/receiver/local_batch.rb

```diff
@@ -25,6 +25,15 @@ def receive!
     # 09/27/11 this is slow
     notify_users
 
+    if @object.respond_to?(:type) && @object.type == "Reshare"
+      root_object = Post.find_by guid: @object.root_guid
+      root_author = Person.find(root_object.author_id)
+      if root_author.local?
+        # create a participation to the user for this reshare, to federate comments to author
+        root_author.owner.participate! @object
+      end
+    end
+
```

Comment:
The method is nicely cleaned up with higher level methods. Let's keep it that way and extract this to a nicely named method, instead of polluting it with lower level code. Keeping the same level of abstraction makes a method easier to follow.